Battery Charging Bug

dguittet opened the pull request:
The calculate_current_for_power_kw function reduces the battery's target power and current to what is available given SOC. Fix the bug in the charging portion where the reduced power and current should have been negative.

Fix #1361

dguittet committed:
fix #1361

brtietz commented:
Thanks for looking into this! A few requests:

1. Can you add a test that establishes this sign convention explicitly? There are a few tests in lib_resilience_test that are close, but I'm getting lost with multiple negative sign conventions. Bonus points for establishing the convention in the discharging function as well.
2. Can you review other uses of the function to ensure the sign convention is correct? I'm seeing some additional inconsistencies in the below.

<img width="1277" height="359" alt="image" src="https://github.com/user-attachments/assets/9fa76e01-638a-4dd9-b277-0ee05458eb88" />

dguittet committed:
Merge branch 'develop' into batt_pwr_fix

dguittet committed:
apply pos/neg limits to calculate_max_charge_kw and calculate_max_discharge_kw

dguittet commented:
Thanks Brian for the good catch. Turns out it was just a few edge cases here and there where `calculate_max_discharge_kw` and `calculate_max_charge_kw` could return values that were outside of their expected bounds so I added limit checks and removed the erroneous sign change in `calculate_current_for_power_kw`

dguittet committed:
revert minor test change

brtietz committed:
Merge branch 'develop' of https://github.com/NREL/ssc into batt_pwr_fix

brtietz reviewed (approved):
Thanks for incorporating the fixes!
